nautobot/core/utils/lookup.py

```diff
@@ -222,11 +222,12 @@ def get_table_for_model(model):
 
     Args:
         model (BaseModel): A model class
+        suffix (str): An additional suffix for the table name (e.g. `Detail`, such as to retrieve `FooDetailTable`)
 
     Returns:
         (Union[Table, None]): Either the `Table` class or `None`
     """
-    return get_related_class_for_model(model, module_name="tables", object_suffix="Table")
+    return get_related_class_for_model(model, module_name="tables", object_suffix=f"{suffix or ''}Table")
```

Comment:
It's non-obvious to me that the `suffix` is actually an infix. I think a suffix of `DetailTable` is more intuitive than a suffix of `Detail` implying `DetailTable`, personally.